gcloud-java-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -144,14 +145,62 @@ public boolean isDone() {
   }
 
   /**
+   * Blocks until this job completes its execution, either failing or succeeding. The job status is
+   * checked every 500 milliseconds. This method returns current job's latest information. If the
+   * job no longer exists, this method returns {@code null}.
+   * <pre> {@code
+   * Job completedJob = job.waitFor();
+   * if (completedJob == null) {
+   *   // job no longer exists
+   * } else if (completedJob.status().error() != null) {
+   *   // job failed, handle error
+   * } else {
+   *   // job completed successfully
+   * }}</pre>
+   *
+   * @throws BigQueryException upon failure
+   * @throws InterruptedException if the current thread gets interrupted while waiting for the job
+   *     to complete
+   */
+  public Job waitFor() throws InterruptedException {
+    return waitFor(500, TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * Blocks until this job completes its execution, either failing or succeeding. The
+   * {@code checkEvery} and {@code unit} parameters determine how often the job's status is checked.
+   * This method returns current job's latest information. If the job no longer exists, this method
+   * returns {@code null}.
+   * <pre> {@code
+   * Job completedJob = job.waitFor(1, TimeUnit.SECONDS);
+   * if (completedJob == null) {
+   *   // job no longer exists
+   * } else if (completedJob.status().error() != null) {
+   *   // job failed, handle error
+   * } else {
+   *   // job completed successfully
+   * }}</pre>
+   *
+   * @throws BigQueryException upon failure
+   * @throws InterruptedException if the current thread gets interrupted while waiting for the job
+   *     to complete
+   */
+  public Job waitFor(int checkEvery, TimeUnit unit) throws InterruptedException {
```

Comment:
I considered it, I preferred overloading because:
- less duplicated code (the vararg option would be very similar both in `BigQuery` and `Compute` and I wasn't sure whether moving it to `gcloud-java-core` given the small use such a class would have)
- I thought about a timeout but it seemed not very useful in this case. And I couldn't think of other options

Vararg option sound good to me so if you have a preference I am happy to give it a try.